confmap/provider/s3provider/provider.go

```diff
@@ -112,27 +113,33 @@ func (*provider) Shutdown(context.Context) error {
 //   - [KEY]    : The key exists in a given bucket, can be used to retrieve a file.
 func s3URISplit(uri string) (string, string, string, error) {
 	// check whether the pattern of s3-uri is correct
-	matched, err := regexp.MatchString(`s3:\/\/(.*)\.s3\.(.*).amazonaws\.com\/(.*)`, uri)
-	if !matched || err != nil {
-		return "", "", "", fmt.Errorf("invalid s3-uri using a wrong pattern")
+	pattern := `^s3:\/\/([a-z0-9\.\-]{3,63})\.s3\.([a-z0-9\-]+).amazonaws\.com\/.`
+	s3Regexp, err := regexp.Compile(pattern)
+	if err != nil {
+		// this error should never happen since the regexp pattern is hard coded
+		return "", "", "", fmt.Errorf("failed to compile the s3 uri-regexp: %q", pattern)
+	}
+
+	matched := s3Regexp.MatchString(uri)
+	if !matched {
+		return "", "", "", fmt.Errorf("s3 uri does not match the pattern: %q", pattern)
 	}
```

Comment:
Since the error should never happen, you can move the regex to an unexported package level variable and use `MustCompile`

Reply:
TIL. thanks!